lib/tests/test_gpg.rs

```diff
@@ -75,6 +75,18 @@ impl GpgEnvironment {
     }
 }
 
+// This will only run the `test` fn if gpg is installed on the system
+fn run_with_guard<F>(test: F)
+where
+    F: FnOnce(),
+{
+    if Command::new("gpgaa").arg("--version").status().is_err() {
+        eprintln!("Skipping because gpg is not installed on the system");
+        return;
+    }
+    test();
+}
+
```

Comment:
nit: alternatively, this could be a declarative macro. I don't know which is better overall. Both approaches have pros and cons.

```rust
macro_rules! gpg_guard {
    () => {
        if Command::new("gpg").arg("--version").status().is_err() {
            eprintln!("Skipping because gpg is not installed on the system");
            return;
        }
    };
}

```

Reply:
Nice! Never used macros before. This looks nice to me will use this. Out of interest what would you consider the pros and cons in this context?

Reply:
closure: + obvious control flow, - nesting
macro: + less noisy, - hidden return statement